base/main_test_bucket_pool.go

```diff
@@ -580,43 +437,44 @@ func (tbp *TestBucketPool) createTestBuckets(numBuckets int, bucketQuotaMB int,
 		// Bucket creation takes a few seconds for each bucket,
 		// so create and wait for readiness concurrently.
 		go func(bucketName string) {
+			ctx := bucketNameCtx(ctx, bucketName)
+
 			tbp.Logf(ctx, "Creating new test bucket")
 			err := tbp.cluster.insertBucket(bucketName, bucketQuotaMB)
 			if err != nil {
 				tbp.Fatalf(ctx, "Couldn't create test bucket: %v", err)
 			}
 
-			namedCollection, defaultCollection, err := tbp.cluster.openTestBucket(tbpBucketName(bucketName), waitForReadyBucketTimeout, tbp.UsingNamedCollections())
-			ctx = updateContextWithKeyspace(ctx, defaultCollection)
+			bucket, err := tbp.cluster.openTestBucket(tbpBucketName(bucketName), waitForReadyBucketTimeout)
 			if err != nil {
 				tbp.Fatalf(ctx, "Timed out trying to open new bucket: %v", err)
 			}
 			openBucketsLock.Lock()
-			openBuckets[bucketName] = openCollections{
-				namedCollection:   namedCollection,
-				defaultCollection: defaultCollection,
-			}
+			openBuckets[bucketName] = bucket
 			openBucketsLock.Unlock()
 
-			ctx = updateContextWithKeyspace(ctx, defaultCollection)
-			// if the bucket is a N1QLStore, clean up prepared statements as-per the advice from the query team.
-			if n1qlStore, ok := AsN1QLStore(defaultCollection); ok {
-				err = n1qlStore.waitUntilQueryServiceReady(time.Minute)
-				if err != nil {
-					tbp.Fatalf(ctx, "Timed out waiting for query service to be ready: %v", err)
-				}
-
-				queryRes, err := n1qlStore.Query(`DELETE FROM system:prepareds WHERE statement LIKE "%`+KeyspaceQueryToken+`%";`, nil, RequestPlus, true)
-				if err != nil {
-					tbp.Fatalf(ctx, "Couldn't remove old prepared statements: %v", err)
+			// If set, the test bucket pool will also create N collections per bucket - rather than just getting the default collection ready.
+			if tbp.useCollections {
+				dynamicDataStore, ok := bucket.(sgbucket.DynamicDataStoreBucket)
+				if !ok {
+					tbp.Fatalf(ctx, "Bucket doesn't support dynamic collection creation")
 				}
-
-				err = queryRes.Close()
-				if err != nil {
-					tbp.Fatalf(ctx, "Failed to close query: %v", err)
+				for i := 0; i < numCollectionsPerBucket; i++ {
+					scopeName := fmt.Sprintf("%s%d", tbpScopePrefix, 0)
+					collectionName := fmt.Sprintf("%s%d", tbpCollectionPrefix, i)
+					ctx := keyspaceNameCtx(ctx, bucketName, scopeName, collectionName)
+					tbp.Logf(ctx, "Creating new collection: %s.%s", scopeName, collectionName)
+					dataStoreName := ScopeAndCollectionName{Scope: scopeName, Collection: collectionName}
+					err := dynamicDataStore.CreateDataStore(dataStoreName)
+					if err != nil {
+						tbp.Fatalf(ctx, "Couldn't create datastore %v.%v: %v", scopeName, collectionName, err)
+					}
+					tbp.emptyPreparedStatements(ctx, bucket.NamedDataStore(dataStoreName))
```

Comment:
Can we safely just remove all prepared statements for the bucket in a single call here, instead of making multiple calls per keyspace?  Looks like we're already doing LIKE %Keyspace% - could just do %BucketName%?